lib/elixir/lib/module.ex

```diff
@@ -725,7 +738,10 @@ defmodule Module do
       end
 
   """
-  def defines?(module, tuple) when is_tuple(tuple) do
+  @spec defines?(module, function_arity) :: boolean
+  def defines?(module, {function_macro_name, arity} = tuple)
+      when is_atom(module) and is_atom(function_macro_name)
+      and is_integer(arity) and arity >= 0 and arity <= 255 do
```

Comment:
Please put the `and` at the end of the previous line, and indent `is_integer` to be aligned with the `is_atom` on the line above:

```elixir
def ...
    when is_atom(module) and is_atom(function_macro_name) and
         is_integer(arity) and arity >= 0 ... do
```